test/spectests/spectests.cpp

```diff
@@ -287,19 +323,33 @@ int main(int argc, char** argv)
 {
     try
     {
-        if (argc < 2)
+        std::string dir;
+        test_settings settings;
+
+        for (auto i = 1; i < argc; ++i)
         {
-            std::cerr << "Missing DIR argument\n";
-            return -1;
+            if (argv[i][0] == '-')
+            {
+                if (argv[i] == std::string{"--skip-validation"})
+                    settings.skip_validation = true;
+                else
+                {
+                    std::cerr << "Unknown argument: " << argv[i] << "\n";
+                    return -1;
+                }
+            }
+            else
+                dir = argv[i];
         }
-        else if (argc > 2)
+
+        if (dir.empty())
         {
-            std::cerr << "Too many arguments\n";
+            std::cerr << "Missing DIR argument\n";
             return -1;
         }
 
-        run_tests_from_dir(argv[1]);
-        return 0;
+        const bool res = run_tests_from_dir(dir, settings);
+        return res ? 0 : -1;
```

Comment:
I think you should return `1` in case of failure. Negative values mean something really exceptional. GTest returns `1` in case of failures.

Reply:
Changed to 1.